irohad/ordering/impl/on_demand_ordering_service_impl.cpp

```diff
@@ -237,11 +237,9 @@ void OnDemandOrderingServiceImpl::packNextProposals(
 }
 
 void OnDemandOrderingServiceImpl::tryErase() {
-  while (round_queue_.size() > number_of_proposals_) {
-    auto &round = round_queue_.front();
-    proposal_map_.erase(round);
-    log_->info("tryErase: erased {}", round);
-    round_queue_.pop();
+  while (proposal_map_.size() > number_of_proposals_) {
```

Comment:
I would rather use the previous logic, where this loop operated only on subset of `proposal_map_` with `map.round < packNextProposals.round`.
I would suggest to first `it = proposal_map_.find(round)`, and then operate on `range(proposal_map_.begin(), it)`.

Reply:
this makes sense. It could be also done by passing current round to `tryErase` and adding condition
```suggestion
  while (proposal_map_.size() > number_of_proposals_ and proposal_map_.begin()->first < current_round) {
```

Reply:
I think that the first condition should only consider the subset, e.g. `size(range(proposal_map_.begin(), it)) > number_of_proposals_` instead of `proposal_map_.size() > number_of_proposals_`.